server/segment_rpc.go

```diff
@@ -133,13 +134,20 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 		Name:  uri,
 	}
 
-	res, err := orch.TranscodeSeg(segData, &hlsStream) // ANGIE - NEED TO CHANGE ALL JOBIDS IN TRANSCODING LOOP INTO STRINGS
+	res, err := orch.TranscodeSeg(segData, &hlsStream)
 
 	// Upload to OS and construct segment result set
 	var segments []*net.TranscodedSegmentData
 	var pixels int64
 	for i := 0; err == nil && i < len(res.TranscodeData.Segments); i++ {
-		name := fmt.Sprintf("%s/%d.ts", segData.Profiles[i].Name, segData.Seq) // ANGIE - NEED TO EDIT OUT JOB PROFILES
+		var ext string
+		ext, err = common.ProfileFormatExtension(segData.Profiles[i].Format)
+		if err != nil {
+			glog.Errorf("Unknown format extension manifestID=%s seqNo=%d err=%s", segData.ManifestID, segData.Seq, err)
+			break
+		}
+		name := fmt.Sprintf("%s/%d%s", segData.Profiles[i].Name, segData.Seq, ext)
+		// The use of := here is probably a bug?!?
```

Comment:
That was the behavior I was expecting, yeah - failure to save should be propagated immediately to the B as an error.

But it looks like this has been here for a long time, and it's actually codified in a [unit test](https://github.com/livepeer/go-livepeer/blob/9659e5de2b63933f20bd34775c748781033b5074/server/segment_rpc_test.go#L337-L401). So I'm uncertain if ignoring the error was meant to address some other case. Best I can think of is something like using the results that were successfully saved, and retrying the rest? But since that's not anything we support right now, I'm also OK correcting the behavior.

Reply:
Bug fixed in https://github.com/livepeer/go-livepeer/pull/1429/commits/9aca289d6e123cdd3541493bc6ee388d36d111a6

Note that we [debit any fees](https://github.com/livepeer/go-livepeer/blob/9aca289d6e123cdd3541493bc6ee388d36d111a6/server/segment_rpc.go#L163-L164) that may have accumulated from an earlier iteration, but we don't return the accompanying pixel counts anymore. Not sure if that'll lead to a problem if the O is used again for subsequent segments (eg, if a refresh occurs soon thereafter). If it seems like this needs more work / thought, I'll drop the commit from this PR and we can address it separately.

Reply:
> Not sure if that'll lead to a problem if the O is used again for subsequent segments (eg, if a refresh occurs soon thereafter)

With the per-stream suspension list, the O wouldn't be reused for awhile so I think this should only be a problem if there are only a small # of available Os (so B can't cycle through other Os before reusing one that returned an error).

That being said, I lean towards debiting fees only if we're returning segments/pixel counts and not debiting fees if we're returning a transcode error to minimize accounting issues (also feels right to not debit fees from a B's balance if O is not returning segments that could actually be used by B). Since this would require moving the `DebitFees()` call and writing some additional test cases I'm cool with dropping the commit and addressing separately.

Reply:
Reverted in 998619c86ce6ba027a18dddbd1931bc6344294d1

Reply:
Sounds good 👍